packages/app-admin-ui/client/providers/AdminMeta.js

```diff
@@ -5,15 +5,29 @@ import React from 'react';
 import List from '../classes/List';
 import { views, readViews, preloadViews } from '../FIELD_TYPES';
 
-const { __pages__: pageViews, ...listViews } = views;
+const { __pages__: pageViews, __hooks__: hookView, ...listViews } = views;
 
 // TODO: Pull this off `window.X` to support server side permission queries
 const { lists, ...srcMeta } = KEYSTONE_ADMIN_META;
 
+const resolveCustomPages = pages => {
+  if (!Array.isArray(pages)) return pages;
+  pages.forEach(page => {
+    if (page.component === 'string') {
+      // this can be simplified once all pages are hooks
+      const [Page] = readViews([pageViews[page.path]]);
```

Comment:
We should batch this readView call with all the other pageViews(and ideally the readViews for the hooks files) so that we don't have a horrible waterfall of requests

Reply:
Can you show me how?